### DISCUSSION TITLE
Empty Givens?

### DISCUSSION THREAD

**Discussion by alexvoss** [Tyler-Keith-Thompson/CucumberSwift#44] [Q&A] [answered]
Hi, 

I have a few things that need to be tested when my app starts and I am not entirely sure how to do this. Am I right to assume that a scenario needs to have at least one each of `Given`, `When`and `Then`? What would I do for `Given` if the `When`step starts the app and there is nothing to do prior? One thing I thought of was to write `Given nothing special` and have a step definition for this that does nothing. Am I holding the wrong end of the stick?

Cheers,

Alex

**Comment by Tyler-Keith-Thompson** [comment, marked as answer]
I think your best bet here is to keep Gherkin as useful as possible

```gherkin
Given some context that does not need to execute
When I run the app
Then some testable outcome is achieved
```

And then have Swift code that matches your Given and simply contains no instructions:

```swift
Given("^some context that does not need to execute$") { _, _ in }
```

Personally, I would add all these steps to a different swift file of some kind so that they don't muddy up code. Examples of how I've gone about that can seen in #23 


HOWEVER, a scenario does *not* need to have the keyword `Given` in it. You could start a scenario with `When` or `Then` because Gherkin has line-based parsing. It has some hierarchy, but not a whole lot. So if your phrasing in Gherkin adequately describes a scenario without a `Given` step, that's totally fine.
